AppWidget/app/src/main/java/com/example/android/appwidget/glance/weather/WeatherWorker.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.appwidget.glance.weather
+
+import android.content.Context
+import androidx.glance.GlanceId
+import androidx.glance.appwidget.GlanceAppWidgetManager
+import androidx.glance.appwidget.state.updateAppWidgetState
+import androidx.glance.appwidget.updateAll
+import androidx.work.*
+import java.time.Duration
+
+class WeatherWorker(
+    private val context: Context,
+    workerParameters: WorkerParameters
+) : CoroutineWorker(context, workerParameters) {
+
+    companion object {
+
+        private val uniqueWorkName = WeatherWorker::class.java.simpleName
+
+        /**
+         * Enqueues a new worker to refresh weather data only if not enqueued already
+         *
+         * Note: if you would like to have different workers per widget instance you could provide
+         * the unique name based on some criteria (e.g selected weather location).
+         *
+         * @param force set to true to replace any ongoing work and expedite the request
+         */
+        fun enqueue(context: Context, force: Boolean = false) {
+            val manager = WorkManager.getInstance(context)
+            val requestBuilder = PeriodicWorkRequestBuilder<WeatherWorker>(
+                Duration.ofMinutes(30)
+            )
+            var workPolicy = ExistingPeriodicWorkPolicy.KEEP
+
+            // Replace any enqueued work and expedite the request
+            if (force) {
+                workPolicy = ExistingPeriodicWorkPolicy.REPLACE
+                requestBuilder.setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
+            }
+
+            manager.enqueueUniquePeriodicWork(
+                uniqueWorkName,
+                workPolicy,
+                requestBuilder.build()
+            )
+        }
+
+        /**
+         * Cancel any ongoing worker
+         */
+        fun cancel(context: Context) {
+            WorkManager.getInstance(context).cancelUniqueWork(WeatherWorker::class.java.simpleName)
```

Comment:
Nit: Maybe use `uniqueWorkName` instead of `WeatherWorker::class.java.simpleName` since you defined it?